File: lib-esphome-client/src/main/java/io/github/arhor/esphome/client/async/EspHomeClientException.java
package io.github.arhor.esphome.client.async;

public abstract class EspHomeClientException extends RuntimeException {

    protected EspHomeClientException(final String message, final Throwable cause) {
        super(message, cause);
    }

    protected EspHomeClientException(final String message) {
        super(message);
    }
}

File: services/lib-esphome-client/src/main/kotlin/io/github/arhor/esphome/client/EspHomeClientException.kt
package io.github.arhor.esphome.client

open class EspHomeClientException(
    message: String,
    cause: Throwable? = null,
) : RuntimeException(message, cause)

class EspHomeTransportException(
    message: String,
    cause: Throwable? = null,
) : EspHomeClientException(message, cause)

class EspHomeProtocolException(
    message: String,
    cause: Throwable? = null,
) : EspHomeClientException(message, cause)

class EspHomeAuthenticationException(
    message: String,
) : EspHomeClientException(message)
